src/screens/AccountNew.js

```diff
@@ -45,7 +45,13 @@ export default class AccountNew extends React.Component {
 }
 
 class AccountNewView extends React.Component {
-  render() {
+
+  shouldComponentUpdate (nextProps) {
+    const hasEmptySeed = nextProps.accounts.state.newAccount.seed === "";
```

Comment:
This looks like a really big code stink to me, even if it solves the problem. `shouldComponentUpdate` should be used to prevent re-renders when relevant props haven't changed, not to be a band-aid over bad state. It's also unclear to me why this is receiving the entire `accounts` store instead of just getting the `newAccount` itself and maybe a callback to inform upstream that account has been selected/created/whatever, separation of concerns is missing here. Anyway, fixing all of that is beyond the scope of this PR I think.

I'd suggest trying to find where the `newAccount.seed` is set to empty string in code, and instead set the entire `newAccount` to `null`, then prevent this component from rendering in the `render` method (it seems to already do that check `if (!selected) {` at line 58).

I think a proper way would be to set the whole `newAccount`

Reply:
The `newAccount` is cleared cleanly [here](https://github.com/paritytech/parity-signer/blob/master/src/stores/AccountsStore.js#L94), right after having been submitted, which seems to be the right place. 

I originally thought there was a race between locking the account [here at componentWillUnmount](https://github.com/paritytech/parity-signer/blob/master/src/screens/AccountBackup.js#L64). which triggers a seed clearing [here in `lockAccount`](https://github.com/paritytech/parity-signer/blob/master/src/stores/AccountsStore.js#L166).

Now another possibility is that we clicked both on the `Done` button and the `enter` key on an Android keyboard at the pin creation step. Both action submit the form [here](https://github.com/paritytech/parity-signer/blob/master/src/screens/AccountPin.js#L124). This allows me to reproduce the problem 100% of the time.

I can't manage to reproduce the problem on master by carefully click only on `Done`, or the `enter` key.

Reply:
The reason why `accounts` is passed and not only `newAccount` is because the components for the `Backup` and `pin` are re-used for "View your recovery phrase" and "Change your pin".